### PR TITLE
Syncer: Implement logic for balance retrieval

### PR DESCRIPTION
**Pull request by TheCharlatan**
Also adds a cli call to trigger the get balance procedure

### CODE REVIEW

**Review by h4sh3d — CHANGES_REQUESTED** [review]
When starting a fresh instance of farcasterd and query a balance (either btc or xmr) we get:
```
[2022-12-13T10:26:08Z WARN  farcaster_node::farcasterd::syncer_state_machine] Request GetBalance(..) from client<2247671496907752132> invalid for state start - invalidating.
[2022-12-13T10:26:08Z INFO  farcaster_node::event] Syncer state machine ended Start -> End
```

This is triggered by

https://github.com/farcaster-project/farcaster-node/blob/bf5365de4d733d8c5083c82335b946e57fc947c9/src/farcasterd/syncer_state_machine.rs#L222-L225

File: src/swapd/runtime.rs [hunk 1]
```diff
@@ -2178,6 +2178,7 @@ impl Runtime {
                                         self.syncer_state.network.into(),
                                         &viewpair,
                                     );
+                                    // safe the monero address to the database now
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/swapd/runtime.rs:2181]: This line is wrong, the monero address is only saved once we have all the private keys aggregated.

[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/swapd/runtime.rs]: 820eadd7ed0637ac2b647c538d7ee91dfbb9b891

File: src/bus/ctl.rs [hunk 2]
```diff
@@ -125,6 +125,8 @@ pub enum CtlMsg {
     #[display("set_address_secret_key")]
     SetAddressSecretKey(AddressSecretKey),
 
+    GetBalance(AddressSecretKey),
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/bus/ctl.rs:129]: 046f77951a92247501b6ac294564f3eb0073d285

**Review by h4sh3d — APPROVED** [review]
tACK 92a48a26a184acb7d28daefdac968adb46871451

**Pull request merged**